packages/truffle-debugger/lib/controller/selectors/index.js

```diff
@@ -48,28 +48,45 @@ const controller = createSelectorTree({
       /**
        * controller.current.location.sourceRange
        */
-      sourceRange: createLeaf([solidity.current.sourceRange], identity),
+      sourceRange: createLeaf(
+        [solidity.current.sourceRange, "/current/loaded"],
+        (range, loaded) => (loaded ? range : null)
+      ),
 
       /**
        * controller.current.location.source
        */
-      source: createLeaf([solidity.current.source], identity),
+      source: createLeaf(
+        [solidity.current.source, "/current/loaded"],
+        (source, loaded) => (loaded ? source : null)
+      ),
 
       /**
        * controller.current.location.node
        */
-      node: createLeaf([solidity.current.node], identity),
+      node: createLeaf(
+        [solidity.current.node, "/current/loaded"],
+        (node, loaded) => (loaded ? node : null)
+      ),
 
       /**
        * controller.current.location.isMultiline
        */
-      isMultiline: createLeaf([solidity.current.isMultiline], identity)
+      isMultiline: createLeaf(
+        [solidity.current.isMultiline, "/current/loaded"],
+        (raw, loaded) => (loaded ? raw : false)
+      )
     },
 
     /**
      * controller.current.finished
      */
-    finished: createLeaf([trace.finished], identity)
+    finished: createLeaf([trace.finished], identity),
+
+    /**
+     * controller.current.loaded
+     */
+    loaded: createLeaf([trace.loaded], identity)
```

Comment:
How do you feel about nesting `finished` and `loaded` into `controller.current.trace`? Seeing `controller.current.loaded` struck me as a bit implicit.

Reply:
Makes sense, I guess.  Can do that.